File: basic math/javascript/is_armstrong.js
const readline = require('readline')

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

const isArmstrong = (num) => {
    let temp = num
    let sum = 0
    while (temp) {
        let digit = temp % 10
        sum += digit ** 3
        temp = parseInt(temp / 10)
    }
    return sum === num
}

rl.question('Enter a number: ', (answer) => {
    let num = parseInt(answer)
    console.log(`${num} is ${isArmstrong(num) ? '' : 'not '}an Armstrong number`)
    rl.close()
})

File: basic math/c++/is_armstrong.cpp
#include <iostream>
using namespace std;

int isArmstrong(int n)
{
    int sum = 0, temp = n;
    while (n > 0)
    {
        int rem = n % 10;
        sum += rem * rem * rem;
        n /= 10;
    }
    return sum == temp;
}

int main()
{
    int n;
    cout << "Enter a number: ";
    cin >> n;
    cout << n << (isArmstrong(n) ? " is an Armstrong number." : " is not an Armstrong number.") << endl;
    return 0;
}